flake.nix

```diff
@@ -336,8 +336,67 @@
               make check-format
               touch $out
             '';
+          template-deps-json = pkgs.runCommand "template-deps-check"
+            {
+              ctlPackageJson = builtins.readFile ./package.json;
+              ctlScaffoldPackageJson = builtins.readFile ./templates/ctl-scaffold/package.json;
+              nativeBuildInputs = [ pkgs.jq ];
+            } ''
+            cd ${self}
+            diff <(jq -S .dependencies <<< $ctlPackageJson) <(jq -S .dependencies <<< $ctlScaffoldPackageJson)
+            diff <(jq -S .devDependencies <<< $ctlPackageJson) <(jq -S .devDependencies <<< $ctlScaffoldPackageJson)
+            touch $out
+          '';
+          template-dhall-diff = pkgs.runCommand "template-dhall-diff-check"
+            (
+              let
+                ctlPackages = import ./spago-packages.nix { inherit pkgs; };
+                ctlScaffoldPackages = import ./templates/ctl-scaffold/spago-packages.nix { inherit pkgs; };
+                conflictOp = acc: depattr: with builtins;
+                  if elem depattr (attrNames ctlScaffoldPackages.inputs) &&
+                  ctlPackages.inputs.${depattr}.version == ctlScaffoldPackages.inputs.${depattr}.version
+                  then acc else [ depattr ] ++ acc;
+                conflicts = builtins.foldl' conflictOp [ ] (builtins.attrNames ctlPackages.inputs);
```

Comment:
It only catches packages that are in ctlPackages, but not in the template, right? We should also catch the inverse case.